pyanaconda/modules/subscription/initialization.py

```diff
@@ -0,0 +1,90 @@
+#
+# Kickstart module for subscription handling.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+RHSM_SYSTEMD_UNIT_NAME = "rhsm.service"
+
+SERVICE_CHECK_INTERVAL = 0.1  # how often to check in s
+
+import time
+
+from dasbus.typing import get_variant, Str
+
+from pyanaconda.core import util
+
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.constants.services import RHSM
+from pyanaconda.modules.common.constants.objects import RHSM_CONFIG
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+class StartRHSMTask(Task):
+    """Task for starting the RHSM DBus service."""
+
+    def __init__(self):
+        super().__init__()
+        self._started_time = None
+
+    @property
+    def name(self):
+        return "Start RHSM DBus service"
+
+    def run(self):
+        """Start the RHSM DBus service.
+
+        And also some related tasks, such as setting RHSM log levels.
+        """
+        # record when the task was started
+        self._start_time = time.time()
+
+        # start the rhsm.service
+        # - this is blocking, but as we are effectively running in a thread
+        # it should not be an issue
+        # - if the return code is non-zero, raise an exception
+        rc = util.start_service(RHSM_SYSTEMD_UNIT_NAME)
+        if rc:
+            return False
+        # create a temporary proxy to set the log levels
+        rhsm_config_proxy = RHSM.get_proxy(RHSM_CONFIG)
+
+        # set RHSM log levels to debug
+        # - otherwise the RHSM log output is not usable for debugging subscription issues
+        log.debug("subscription: setting RHSM log level to DEBUG")
+        rhsm_config_proxy.Set("logging.default_log_level", get_variant(Str, "DEBUG"), "")
+        # all seems fine
+        return True
+
+    def is_service_available(self, timeout, check_interval=SERVICE_CHECK_INTERVAL):
+        """Return if RHSM service is available or wait if startup is ongoing."""
+        while self.is_running:
```

Comment:
Active waiting is not necessary. You can get the running thread with `thread = threadMgr.get(self._thread_name)` and wait with a timeout with `thread.join(timeout)`.

See: https://docs.python.org/3.9/library/threading.html#threading.Thread.join

Reply:
Oh, right - that'c clever. :)

BTW, I wonder if we might want to implement a mechanism like this in the Task itself in the future ? Seems useful.